packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx

```diff
@@ -0,0 +1,227 @@
+
+    const handleSubmit = useCallback(async () => {
+        if (!recoveryPhrase) {
+            throw new Error('Missing recovery phrase');
+        }
+        if (!network) {
+            throw new Error('Network is not specified');
+        }
+        if (!credential) {
+            throw new Error('credential is not loaded');
+        }
+
+        const net = getNet(network);
+
+        const global = await getGlobal(client);
+        const wallet = ConcordiumHdWallet.fromHex(recoveryPhrase, net);
+
+        const requestStatement = { statement: [statement], id: createAccountDID(net, credential?.credId) };
+
+        const commitmentInput = getAccountCredentialCommitmentInput(
+            requestStatement,
+            wallet,
+            identities.value,
+            credentials.value
+        );
+
+        const request = {
+            challenge,
+            credentialStatements: [requestStatement],
+        };
+
+        return proveRequest(request, [commitmentInput], global);
+    }, [credential, identity, recoveryPhrase, network]);
+
+    useEffect(() => onClose(onReject), [onClose, onReject]);
+
+    const description = useMemo(() => {
+        if (!age) {
+            return '';
+        }
+        const { ageMin, ageMax } = age;
+
+        if (ageMax === undefined) {
+            return t('ageMin', { age: ageMin });
+        }
+
+        if (ageMin === undefined) {
+            return t('ageMax', { age: ageMax });
+        }
+
+        if (ageMin === ageMax) {
+            return t('ageExact', { age: ageMin });
+        }
+
+        return t('ageBetween', { ageMin, ageMax });
+    }, [age]);
+
+    if (loadingPasscode) {
+        // This will be near instant, as we're just waiting for the Chrome async store
+        return null;
+    }
+
+    if (!sessionPasscode) {
+        return <Navigate to={absoluteRoutes.login.path} state={{ to: -1 }} />;
+    }
+
+    if (more) {
+        return <Web3ProofRequest onSubmit={onSubmit} onReject={onReject} />;
+    }
+
+    return (
+        <div className="age-proof__request">
+            <Toast />
+            <div>
+                <div className="age-proof__logos">
+                    <Logo className="age-proof__logo" />
+                </div>
+                <h3 className="m-t-40 text-center age-proof__description">{description}</h3>
+                <Button clear onClick={() => setMore(true)} className="age-proof__more-details">
+                    {t('moreDetails', { dappName })}
+                </Button>
+            </div>
+            <Button
+                className="flex-center age-proof__verify-button m-20"
+                onClick={() => {
+                    setCreatingProof(true);
+                    handleSubmit()
+                        .then(withClose(onSubmit))
+                        .catch((e) => {
+                            setCreatingProof(false);
+                            logError(e);
+                            // TODO what to do
```

Comment:
It depends on what can cause this to fail.

Reply:
I just "improved the message" and left it at that.